### PR TITLE
Remove Device mod, use ClimaComms

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR removes the `Device` module, and uses ClimaComms. xref: https://github.com/CliMA/ClimaCore.jl/pull/1214#issuecomment-1531941357

### CODE REVIEW

**Review by valeriabarra — APPROVED** [review]

**Comment by simonbyrne** [comment]
we should add a deprecation

**Comment by charleskawczynski** [comment]
> we should add a deprecation

Does that mean we need to keep the module around?

**Comment by simonbyrne** [comment]
maybe create a file `src/deprecated.jl`, and put the module and function definitions in there (once you have this working)

**Comment by charleskawczynski** [comment]
@simonbyrne, how does that look?

**Comment by charleskawczynski** [comment]
bors try

File: src/deprecated.jl [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Device module
+module Device
+using ClimaComms
+using CUDA
+
+Base.@deprecate device(; disablegpu = false) ClimaComms.device(;
+    disablegpu = false,
+) false
```
Comment (simonbyrne) [inline src/deprecated.jl:8]: ClimaComms.device doesn't take a `disablegpu` arg.
```suggestion
Base.@deprecate device(; disablegpu = false) if disablegpu
    ClimaComms.CPUDevice()
else
    ClimaComms.device()
end false
```

Comment (charleskawczynski) [reply src/deprecated.jl:8]: Ah, whoops, I thought I had looked at the definition in ClimaComms, must have been in ClimaCore. Thanks!

**Comment by charleskawczynski** [comment]
bors try

**Review by simonbyrne — APPROVED** [review]

**Comment by charleskawczynski** [comment]
( I canceled the try once this was approved and seemed to be working )

**Pull request merged**